[quant][bc-breaking] Remove deprecated QConfigDynamic

andrewor14 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111673

Summary: QConfigDynamic was deprecated in PyTorch 1.12. It has
continued to cause confusion to users who wish to use dynamic
quantization. This commit removes this deprecated API and
requires users to use QConfig instead.

BC-breaking before:
```
qconfig = QConfigDynamic(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

BC-breaking after:
```
qconfig = QConfig(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

Test Plan:
python test/test_quantization.py

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Differential Revision: [D50697496](https://our.internmc.facebook.com/intern/diff/D50697496)

andrewor14 committed:
[quant][bc-breaking] Remove deprecated QConfigDynamic

Summary: QConfigDynamic was deprecated in PyTorch 1.12. It has
continued to cause confusion to users who wish to use dynamic
quantization. This commit removes this deprecated API and
requires users to use QConfig instead.

BC-breaking before:
```
qconfig = QConfigDynamic(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

BC-breaking after:
```
qconfig = QConfig(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

Test Plan:
python test/test_quantization.py

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

[ghstack-poisoned]

andrewor14 commented:
@jerryzh168 @supriyar @HDCharles Do you think this is a good time to remove this API or is it too early still?

jerryzh168 commented:
yeah I think we can remove now

jerryzh168 reviewed (approved):
please make sure to import to internal and check sandcastle as well

andrewor14 commented:
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50697496).

jerryzh168 commented:
@andrewor14 should we land this?

andrewor14 commented:
Yeah let me update this

andrewor14 commented:
@pytorchbot rebase

andrewor14 committed:
Update on "[quant][bc-breaking] Remove deprecated QConfigDynamic"


Summary: QConfigDynamic was deprecated in PyTorch 1.12. It has
continued to cause confusion to users who wish to use dynamic
quantization. This commit removes this deprecated API and
requires users to use QConfig instead.

BC-breaking before:
```
qconfig = QConfigDynamic(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

BC-breaking after:
```
qconfig = QConfig(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

Test Plan:
python test/test_quantization.py

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Differential Revision: [D50697496](https://our.internmc.facebook.com/intern/diff/D50697496)

[ghstack-poisoned]

andrewor14 commented:
@pytorchbot rebase

andrewor14 committed:
Update on "[quant][bc-breaking] Remove deprecated QConfigDynamic"


Summary: QConfigDynamic was deprecated in PyTorch 1.12. It has
continued to cause confusion to users who wish to use dynamic
quantization. This commit removes this deprecated API and
requires users to use QConfig instead.

BC-breaking before:
```
qconfig = QConfigDynamic(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

BC-breaking after:
```
qconfig = QConfig(
    activation=default_dynamic_quant_observer,
    weight=default_weight_observer,
)
```

Test Plan:
python test/test_quantization.py

Reviewers: jerryzh168

Subscribers: jerryzh168, supriyar

Differential Revision: [D50697496](https://our.internmc.facebook.com/intern/diff/D50697496)

[ghstack-poisoned]
